### PR TITLE
update example in classification losses

### PR DESCRIPTION
**Pull request by rohitgr7**
Just updated a few examples that were either failing or raising deprecated warnings.

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1141,7 +1141,7 @@ class probabilities only when a single class label per minibatch item is too res
         >>>
         >>> # Example of target with class probabilities
         >>> input = torch.randn(3, 5, requires_grad=True)
-        >>> target = torch.randn(3, 5).softmax(dim=1)
+        >>> target = torch.randn(3, 5).softmax(dim=1).argmax(dim=1)
```
Comment (albanD) [inline torch/nn/modules/loss.py:1144]: This is expected. The example above shows a target with indices and this one shows the soft target version.

File: torch/nn/functional.py [hunk 2]
```diff
@@ -2933,7 +2933,7 @@ def cross_entropy(
         >>>
         >>> # Example of target with class probabilities
         >>> input = torch.randn(3, 5, requires_grad=True)
-        >>> target = torch.randn(3, 5).softmax(dim=1)
+        >>> target = torch.randn(3, 5).softmax(dim=1).argmax(dim=1)
```
Comment (albanD) [inline torch/nn/functional.py:2936]: Same as for the one below, this is expected as this example shows the soft label version.

**Review by albanD — COMMENTED** [review]
Thanks for the update in binary cross entropy.
The other ones should be removed though I think.

File: torch/nn/functional.py [hunk 3]
```diff
@@ -2991,9 +2991,9 @@ def binary_cross_entropy(
 
     Examples::
 
-        >>> input = torch.randn((3, 2), requires_grad=True)
-        >>> target = torch.rand((3, 2), requires_grad=False)
-        >>> loss = F.binary_cross_entropy(F.sigmoid(input), target)
+        >>> input = torch.randn(3, 2, requires_grad=True)
+        >>> target = torch.rand(3, 2, requires_grad=False)
+        >>> loss = F.binary_cross_entropy(torch.sigmoid(input), target)
```
Comment (jbschlosser) [inline torch/nn/functional.py:2996]: I think `F.sigmoid` is preferred here to match `F.binary_cross_entropy`

File: torch/nn/functional.py [hunk 4]
```diff
@@ -2991,9 +2991,9 @@ def binary_cross_entropy(
 
     Examples::
 
-        >>> input = torch.randn((3, 2), requires_grad=True)
-        >>> target = torch.rand((3, 2), requires_grad=False)
-        >>> loss = F.binary_cross_entropy(F.sigmoid(input), target)
+        >>> input = torch.randn(3, 2, requires_grad=True)
+        >>> target = torch.rand(3, 2, requires_grad=False)
```
Comment (jbschlosser) [inline torch/nn/functional.py:2995]: This change looks good :)

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! Agree with @albanD that we only want the changes for BCE.

File: torch/nn/modules/loss.py (hunk 1 above)
Comment (rohitgr7) [reply torch/nn/modules/loss.py:1144]: ah! thanks.. my bad.. didn't check the env. was using an old version.

Comment (rohitgr7) [inline torch/nn/modules/loss.py:1144]:
```suggestion
        >>> target = torch.randn(3, 5).softmax(dim=1)
```

File: torch/nn/functional.py (hunk 2 above)
Comment (rohitgr7) [inline torch/nn/functional.py:2936]:
```suggestion
        >>> target = torch.randn(3, 5).softmax(dim=1)
```

**Review by albanD — APPROVED** [review]
Thanks!
